src/engine-actions/create-channels.js

```diff
@@ -0,0 +1,186 @@
+const {
+  connectPeer,
+  openChannel
+} = require('../lnd-actions')
+const {
+  Big,
+  networkAddressFormatter,
+  loggablePubKey
+} = require('../utils')
+const getUncommittedBalance = require('./get-uncommitted-balance')
+const { CHANNEL_ROUNDING } = require('../constants')
+
+/**
+ * Determine the amount to fund channels with by taking into account
+ * uneconomic channels (that are too small to be worth it.)
+ * @param   {Engine} engine
+ * @param   {string} fundingAmount - int64 string of amount desired to open
+ * @param   {number} roundBehavior - Behavior to use when encoutnering a channel that is too small.
+ * @returns {Big}                    Modified amount to open in channels
+ * @private
+ */
+function getAmountForChannels (engine, fundingAmount, roundBehavior) {
+  const {
+    maxChannelBalance,
+    minChannelBalance,
+    quantumsPerCommon,
+    symbol,
+    logger
+  } = engine
+
+  let amount = Big(fundingAmount)
+
+  // our last channel will be our only channel not at maxChannelBalance, so we need to handle it separately
```

Comment:
Nice comment, this is much easier to understand how the next chunk of logic should work without context.